src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -106,23 +107,27 @@ public static partial class JsonSerializer
         {
             Debug.Assert(jsonPropertyInfo != null);
 
-            IDictionary? extensionData = (IDictionary?)jsonPropertyInfo.GetValueAsObject(obj);
+            var extensionData = jsonPropertyInfo.GetValueAsObject(obj);
```

Comment:
```suggestion
            object? extensionData = jsonPropertyInfo.GetValueAsObject(obj);
```

nit: it's not clear what the type of this variable is: see  [rule 10](https://github.com/dotnet/runtime/blob/6d9e5035905ae71b5fceeb8bb5a2c01f7622b7f6/docs/coding-guidelines/coding-style.md).

Reply:
applied in 9884c29